zproject/urls.py

```diff
@@ -194,6 +194,9 @@
     url(r'^messages/matches_narrow$', rest_dispatch,
         {'GET': 'zerver.views.messages.messages_in_narrow_backend'}),
 
+    url(r'^messages/delete_in_topic$', rest_dispatch,
+        {'POST': 'zerver.views.messages.delete_in_topic'}),
+
```

Comment:
Let's do `^streams/(?P<stream_id>\d+)/delete_topic$` as the URL, and put the code in zerver.views.streams, next to `get_topics_backend`.